### PR TITLE
Add unit tests for `Settings`.

### PR DESCRIPTION
**Pull request by dector**
Covered `Settings` with unit-tests.

### Pull Request checklist
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] ~~**Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one~~
- [x] ~~**Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features~~

### CODE REVIEW

File: app/src/test/java/org/mozilla/fenix/utils/SettingsTest.kt [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+    @Test
+    fun usePrivateMode() {
+        assertFalse(settings.usePrivateMode)
+
+        settings.setPrivateMode(true)
+        assertTrue(settings.usePrivateMode)
+
+        settings.setPrivateMode(false)
+        assertFalse(settings.usePrivateMode)
```
Comment (jhugman) [inline app/src/test/java/org/mozilla/fenix/utils/SettingsTest.kt]: This API is pretty grim/Java-y. This should not be fixed in this PR :)

Comment (dector) [reply app/src/test/java/org/mozilla/fenix/utils/SettingsTest.kt]: Do you mean api of the component or test style?

Comment (dector) [reply app/src/test/java/org/mozilla/fenix/utils/SettingsTest.kt]: Anyway, I usually use `Given-When-Then` comments to improve readability. Will add 'em here as well.

**Comment by boek** [comment]
Thank you @dector!

**Pull request merged**